### DISCUSSION TITLE
How to intercept SOAP request before reaching the service class

### DISCUSSION THREAD

**Discussion by robertmircea** [CoreWCF/CoreWCF#1173] [Q&A]
I would like to insert a logger.BeginScope(<name of the soap action>) with the name of the SOAPAction which is about to be executed in order to have contextual  information about the action for all my logs generated during that request. 

What would be a go place to have intercept code BEFORE execution flow reaches my service method corresponding to that SOAPAction?

**Comment by dnperfors** [comment]
The only way I found is by using a ServiceBehavior as described in #619. By using the HttpContext you can request an ILogger and use `BeginScope()` on that.

However, I didn't find a way to let this value "stick" so that it is used in logging by other middleware.
